bodhi/server/ffmarkdown.py

```diff
@@ -53,25 +53,29 @@ def bug_url(tracker, idx):
 
     Args:
         tracker (basestring): Which bug tracker is being referenced. May be any of 'fedora',
-            'gnome', 'kde', 'mozilla', 'pear', 'perl', 'php', 'python', 'rh', or 'rhbz'.
+            'gcc', 'gnome', 'kde', 'mozilla', 'pear', 'perl', 'php', 'python', 'rh', 'rhbz'
+            or 'sourceware'.
         idx (basestring or int): The bug number.
     Returns:
         basestring: The URL of the given bug.
     Raises:
         KeyError: If the given tracker is not supported by this function.
     """
     try:
-        return {
-            'fedora': "https://bugzilla.redhat.com/show_bug.cgi?id=%s",
-            'gnome': "https://bugzilla.gnome.org/show_bug.cgi?id=%s",
-            'kde': "https://bugs.kde.org/show_bug.cgi?id=%s",
-            'mozilla': "https://bugzilla.mozilla.org/show_bug.cgi?id=%s",
-            'pear': "http://pear.php.net/bugs/bug.php?id=%s",
-            'perl': "https://rt.cpan.org/Public/Bug/Display.html?id=%s",
-            'php': "https://bugs.php.net/bug.php?id=%s",
-            'python': "https://bugs.python.org/issue%s",
-            'rh': "https://bugzilla.redhat.com/show_bug.cgi?id=%s",
-            'rhbz': "https://bugzilla.redhat.com/show_bug.cgi?id=%s"}[tracker.lower()] % idx
+        return (
+            {'fedora': "https://bugzilla.redhat.com/show_bug.cgi?id=%s",
+             'gcc': "https://gcc.gnu.org/bugzilla/show_bug.cgi?id=%s",
+             'gnome': "https://bugzilla.gnome.org/show_bug.cgi?id=%s",
+             'kde': "https://bugs.kde.org/show_bug.cgi?id=%s",
+             'mozilla': "https://bugzilla.mozilla.org/show_bug.cgi?id=%s",
+             'pear': "https://pear.php.net/bugs/bug.php?id=%s",
+             'perl': "https://rt.cpan.org/Public/Bug/Display.html?id=%s",
+             'php': "https://bugs.php.net/bug.php?id=%s",
+             'python': "https://bugs.python.org/issue%s",
+             'rh': "https://bugzilla.redhat.com/show_bug.cgi?id=%s",
+             'rhbz': "https://bugzilla.redhat.com/show_bug.cgi?id=%s",
+             'sourceware': "https://sourceware.org/bugzilla/show_bug.cgi?id=%s", }
+            [tracker.lower()] % idx)
```

Comment:
Let's not alter the code style here - we can add just the two new trackers.

Reply:
Ah, I see that we do need to alter it a bit because the line goes too long.

I think I'll just make the dictionary be separate of the return statement to make it more readable and also keep the line short enough.

Reply:
I'll add this:

```
diff --git a/bodhi/server/ffmarkdown.py b/bodhi/server/ffmarkdown.py
index 88465112..5d157f3d 100644
--- a/bodhi/server/ffmarkdown.py
+++ b/bodhi/server/ffmarkdown.py
@@ -62,20 +62,21 @@ def bug_url(tracker, idx):
         KeyError: If the given tracker is not supported by this function.
     """
     try:
-        return (
-            {'fedora': "https://bugzilla.redhat.com/show_bug.cgi?id=%s",
-             'gcc': "https://gcc.gnu.org/bugzilla/show_bug.cgi?id=%s",
-             'gnome': "https://bugzilla.gnome.org/show_bug.cgi?id=%s",
-             'kde': "https://bugs.kde.org/show_bug.cgi?id=%s",
-             'mozilla': "https://bugzilla.mozilla.org/show_bug.cgi?id=%s",
-             'pear': "http://pear.php.net/bugs/bug.php?id=%s",
-             'perl': "https://rt.cpan.org/Public/Bug/Display.html?id=%s",
-             'php': "https://bugs.php.net/bug.php?id=%s",
-             'python': "https://bugs.python.org/issue%s",
-             'rh': "https://bugzilla.redhat.com/show_bug.cgi?id=%s",
-             'rhbz': "https://bugzilla.redhat.com/show_bug.cgi?id=%s",
-             'sourceware': "https://sourceware.org/bugzilla/show_bug.cgi?id=%s", }
-            [tracker.lower()] % idx)
+        trackers = {
+            'fedora': "https://bugzilla.redhat.com/show_bug.cgi?id=%s",
+            'gcc': "https://gcc.gnu.org/bugzilla/show_bug.cgi?id=%s",
+            'gnome': "https://bugzilla.gnome.org/show_bug.cgi?id=%s",
+            'kde': "https://bugs.kde.org/show_bug.cgi?id=%s",
+            'mozilla': "https://bugzilla.mozilla.org/show_bug.cgi?id=%s",
+            'pear': "http://pear.php.net/bugs/bug.php?id=%s",
+            'perl': "https://rt.cpan.org/Public/Bug/Display.html?id=%s",
+            'php': "https://bugs.php.net/bug.php?id=%s",
+            'python': "https://bugs.python.org/issue%s",
+            'rh': "https://bugzilla.redhat.com/show_bug.cgi?id=%s",
+            'rhbz': "https://bugzilla.redhat.com/show_bug.cgi?id=%s",
+            'sourceware': "https://sourceware.org/bugzilla/show_bug.cgi?id=%s"}
+
+        return trackers[tracker.lower()] % idx
 
     except KeyError:
         return None
```